src/Microsoft.Security.Utilities.Core/ISecretMasker.cs

```diff
@@ -9,6 +9,6 @@ namespace Microsoft.Security.Utilities;
 
 public interface ISecretMasker
 {
-    ICollection<Detection> DetectSecrets(string input);
```

Comment:
>ICollection [](http://example.com/codeflow?start=4&length=11)

ICollection returns a count. This interface is contrary to implementing yield iterators, which are currently deeply embedded in our design patterns.